pyaedt/edb_core/materials.py

```diff
@@ -0,0 +1,128 @@
+from __future__ import absolute_import  # noreorder
+
+import logging
+import warnings
+
+logger = logging.getLogger(__name__)
+
+
+class Material:
+    """Manages EDB methods for material property management."""
+
+    def __init__(self, pclass, edb_material_def):
+        self._pclass = pclass
+        self._name = edb_material_def.GetName()
+        self._edb_material_def = edb_material_def
+
+    def _edb_value(self, value):
+        return self._pclass._edb_value(value)
+
+    @property
+    def name(self):
+        """Retrieve material name."""
+        return self._name
+
+    @property
+    def _db(self):
+        return self._pclass._db
+
+    @property
+    def _edb(self):
+        return self._pclass._edb
+
+    @property
+    def conductivity(self):
+        material_id = self._edb.Definition.MaterialPropertyId.Conductivity
+        _, value = self._edb_material_def.GetProperty(material_id)
+        return value.ToDouble()
+
+    @conductivity.setter
+    def conductivity(self, value):
+        """Retrieve material conductivity."""
+        material_id = self._edb.Definition.MaterialPropertyId.Conductivity
+        self._edb_material_def.SetProperty(material_id, self._edb_value(value))
+
+    @property
+    def permittivity(self):
+        """Retrieve material permittivity."""
+        material_id = self._edb.Definition.MaterialPropertyId.Permittivity
+        _, value = self._edb_material_def.GetProperty(material_id)
+        return value.ToDouble()
+
+    @permittivity.setter
+    def permittivity(self, value):
+        material_id = self._edb.Definition.MaterialPropertyId.Permittivity
+        self._edb_material_def.SetProperty(material_id, self._edb_value(value))
+
+    @property
+    def loss_tangent(self):
+        """Retrieve material loss tangent."""
+        material_id = self._edb.Definition.MaterialPropertyId.DielectricLossTangent
+        _, value = self._edb_material_def.GetProperty(material_id)
+        return value.ToDouble()
+
+    @loss_tangent.setter
+    def loss_tangent(self, value):
+        material_id = self._edb.Definition.MaterialPropertyId.DielectricLossTangent
+        self._edb_material_def.SetProperty(material_id, self._edb_value(value))
+
+    @property
+    def thermal_conductivity(self):
+        """Retrieve material thermal conductivity."""
+        material_id = self._edb.Definition.MaterialPropertyId.ThermalConductivity
+        _, value = self._edb_material_def.GetProperty(material_id)
+        return value.ToDouble()
+
```

Comment:
@ring630 
Just to be sure and for the confirmation, it is not possible to set the `thermal_conductivity`? Is it why we don't have a setter for this property?